docs/interactions.md

```diff
@@ -1,310 +1,372 @@
 
-    Charlie L1  TokenX.approve(RealityETH, 2000000)
-    Charlie L1  RealityETH.submitAnswer(gov_question_id, 1, 2000000)
+    Charlie L1.GovToken.approve(RealityETH, 2000000)
+    Charlie L1.RealityETH.submitAnswer(upgrade_question_id, 1, 2000000)
 ```
-Next step:
-* Upgrade question finalizes as 1? [Execute a governance change](#execute-a-governance-change)
-* Upgrade question finalizes as 0? No need to do anything
-* May be contested: [Challenge an arbitration result](#challenge-an-arbitration-or-governance-result)
 
-### Finalize a completed governance change
-```
-    Charlie L1  RealityETH.finalizeQuestion(gov_question_id)
-```
-Next step: 
-* [Execute a governance change](execute-a-governance-change)
+#### Next step:
 
-NB On the normal non-forkable version of Reality.eth finalization happens automatically without a transaction. 
-It's its own transaction on the forkable version because forking for one question affects whether others can finalize, even within the same timestamp.
+ * Upgrade question finalizes as 1? [Execute a contract upgrade](#execute-a-contract-upgrade)
+ * Upgrade question finalizes as 0? No need to do anything
+ * May be contested: [Challenge a contract upgrade](#challenge-a-contract-upgrade)
 
-### Propose an urgent governance change
+### Execute a contract upgrade
 ```
-    Charlie L1  ForkManager.beginUpgradeBridge
-        gov_question_id = RealityETH.askQuestion("should we freeze exits and upgrade to ForkManager XYZ?")
-    Charlie L1  TokenX.approve(RealityETH, 2000000)
-    Charlie L1  RealityETH.submitAnswer(gov_question_id, 1, 2000000)
-    Charlie L1  ForkManager.freezeBridges(gov_question_id)
-                    RealityETH.getBestAnswer(gov_question_id)
-                    RealityETH.getBond(gov_question_id)
-                    # Update self to say there are no available bridges
+    Charlie L1.ForkableRollupGovernor.executeProposition(upgrade_question_id, to, bytecode) 
+                RealityETH.resultFor(upgrade_question_id)
+                call(to, bytecode)
+    (TODO: Handle freezing of bridges etc if the proposition is urgent and there is a high bond to that effect)
 ```
-* Upgrade question finalizes as 1? [Execute a governance change](#execute-a-governance-change)
-* Upgrade question finalizes as 0? [Clear a failed urgent governance proposal](#clear-a-failed-urgent-governance-proposal)
-* May be contested: [Challenge an arbitration result](#challenge-an-arbitration-or-governance-result)
 
-### Execute a governance change
+### Challenge a contract upgrade
 ```
-    Charlie L1  ForkManager.executeBridgeUpdate(gov_question_id) 
-                    RealityETH.resultFor(gov_question_id)
-                    # Update to reflect child forkmanager
-                    # Has the effect of unfreezing bridges, may be new bridges
+    Bob     L1.GovToken.approve(L1ForkArbitrator, fork_fee)
```

Comment:
what happens if there are several questions in parallel? Do we need this?
(not sure whether we want to answer such detailed questions in the spec...). But I think this was one open question

Reply:
Any parallel question is frozen when fork-arbitration is initialized, then imported to a new reality.eth on each child chain. I'll add this the import to the doc.

Reply:
Added in 35543aefc509349b26d795f632d7422a97f7c9ee